src/compiler/checker.ts

```diff
@@ -25858,7 +25858,9 @@ namespace ts {
                 case SyntaxKind.ParenthesizedExpression: {
                     // Like in `checkParenthesizedExpression`, an `/** @type {xyz} */` comment before a parenthesized expression acts as a type cast.
                     const tag = isInJSFile(parent) ? getJSDocTypeTag(parent) : undefined;
-                    return tag ? getTypeFromTypeNode(tag.typeExpression.type) : getContextualType(parent as ParenthesizedExpression, contextFlags);
+                    return !tag ? getContextualType(parent as ParenthesizedExpression, contextFlags) :

```

Comment:
I inverted the order here so that we have one condition per branch (i.e., `!a ? x : b ? y : z`) instead of nested conditions (i.e., `a ? b ? y : x : z`).